include/morphio/enums.h

```diff
@@ -2,63 +2,68 @@
+ * See morphio::mut::modifiers for more information **/
 enum Option {
-    NO_MODIFIER = 0x00,
-    TWO_POINTS_SECTIONS = 0x01,
-    SOMA_SPHERE = 0x02,
-    NO_DUPLICATES = 0x04,
-    NRN_ORDER = 0x08
+    NO_MODIFIER = 0x00,          //!< Read morphology as is without any modification
+    TWO_POINTS_SECTIONS = 0x01,  //!< Read sections only with 2 or more points
+    SOMA_SPHERE = 0x02,          //!< Interpret morphology soma as a sphere
+    NO_DUPLICATES = 0x04,        //!< Skip duplicating points
+    NRN_ORDER = 0x08             //!< Order of neurites will be the same as in NEURON simulator
 };
 
 /**
-   This enum should be kept in sync with the warnings
-   defined in ErrorMessages.
-   It is used to define which warnings are ignored by the user
+   All possible warnings that can be ignored by user. This enum should be kept in sync with the
+   warnings defined in morphio::ErrorMessages.
 **/
 enum Warning {
-    UNDEFINED,
-    MITOCHONDRIA_WRITE_NOT_SUPPORTED,
-    WRITE_NO_SOMA,
-    SOMA_NON_CONFORM,
-    NO_SOMA_FOUND,
-    DISCONNECTED_NEURITE,
-    WRONG_DUPLICATE,
-    APPENDING_EMPTY_SECTION,
-    WRONG_ROOT_POINT,
-    ONLY_CHILD,
-    WRITE_EMPTY_MORPHOLOGY,
-    ZERO_DIAMETER
+    UNDEFINED,                         //!< undefined value
+    MITOCHONDRIA_WRITE_NOT_SUPPORTED,  //!< Mitochondria can be saved in H5 format only
+    WRITE_NO_SOMA,                     //!< Writing without a soma
+    SOMA_NON_CONFORM,      //!< Soma does not conform the three point soma spec from NeuroMorpho.org
+    NO_SOMA_FOUND,         //!< No soma found in a file
+    DISCONNECTED_NEURITE,  //!< Found a disconnected neurite in a morphology
+    WRONG_DUPLICATE,       //!< A wrong duplicate point in a section
+    APPENDING_EMPTY_SECTION,  //!< Appending of an empty section
+    WRONG_ROOT_POINT,         //!< A wrong root point of a neurite in the case of 3 points soma
+    ONLY_CHILD,               //!< Single child sections are not allowed in SWC format
+    WRITE_EMPTY_MORPHOLOGY,   //!< Writing empty morphology
+    ZERO_DIAMETER             //!< Zero section diameter
 };
 
 enum AnnotationType {
     SINGLE_CHILD,
 };
 
 /** The cell family represented by morphio::Morphology. */
-enum CellFamily { NEURON = 0, GLIA = 1, SPINE = 2 };
+enum CellFamily {
+    NEURON = 0,  //!< Neuron
+    GLIA = 1,    //!< Glia
+    SPINE = 2    //!< Spine
+};
 
+/** Soma type. */
 enum SomaType {
-    SOMA_UNDEFINED = 0,
-    SOMA_SINGLE_POINT,
-    SOMA_NEUROMORPHO_THREE_POINT_CYLINDERS,
-    SOMA_CYLINDERS,
-    SOMA_SIMPLE_CONTOUR
+    SOMA_UNDEFINED = 0,                      //!< Undefined soma
+    SOMA_SINGLE_POINT,                       //!< Single point soma
+    SOMA_NEUROMORPHO_THREE_POINT_CYLINDERS,  //!< Soma made of three cylinders
+    SOMA_CYLINDERS,                          //!< Soma made of cylinders
+    SOMA_SIMPLE_CONTOUR                      //!< Contour soma. Assumed that contour is in XY plane
```

Comment:
A two-dimensional soma, then?

Reply:
```suggestion
    SOMA_SIMPLE_CONTOUR                      //!< Contour(two-dimensional) soma. Assumed that contour is in XY plane
```

Reply:
Was asking to clarify… doesn't seem like a two-dimensional Soma makes sense. Is it like a cylinder, but with the contour as a cross-section rather than a circle?

Reply:
I don't know more than in source files, and I don't see that this type of Soma is actually set. Please advice if you know something.

Reply:
Unfortunately, I don't :(